hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketAcceptorThread.java

```diff
@@ -77,46 +92,93 @@ public void run() {
             logger.finest("Starting SocketAcceptor on " + serverSocketChannel);
         }
 
-        Selector selector = null;
+        if (selectorWorkaround) {
```

Comment:
Logging the selector mode once in `NonBlockingIOThreadingModel.start` should be sufficient, will remove this log entry (https://github.com/vbekiaris/hazelcast/blob/master-nio-workaround/hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingIOThreadingModel.java#L125-L125).